### PR TITLE
BUILD: CMake fixes

### PR DESCRIPTION
**Pull request by p01arst0rm**
Collection of CMake fixes that can be bundled together.
Unifying code style, moved openal into its own 
directory structure. Added missing license file for 
stblib headers.

### CODE REVIEW

**Comment by p01arst0rm** [comment]
cc @hcorion

File: 3rdparty/discord-rpc/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# discord_rpc
+add_library(3rdparty_discord-rpc INTERFACE)
```
Comment (hcorion) [inline 3rdparty/discord-rpc/CMakeLists.txt:2]: Probably a good idea to be semi-consistent with upstream Discord RPC's official CMakeLists.txt https://github.com/discord/discord-rpc/blob/master/CMakeLists.txt
And change this to DiscordRPC

File: 3rdparty/stblib/LICENSE [hunk 2]
```diff
@@ -0,0 +1,37 @@
+This software is available under 2 licenses -- choose whichever you prefer.
```
Comment (hcorion) [inline 3rdparty/stblib/LICENSE:1]: It feels kinda weird to list both licenses in a downstream repository like this, I'm guessing we're actually supposed to pick one. We can just go with Alternative A.

**Review by hcorion — CHANGES_REQUESTED** [review]

File: 3rdparty/stblib/LICENSE (hunk 2 above)
Comment (p01arst0rm) [reply 3rdparty/stblib/LICENSE:1]: okeydoke

File: 3rdparty/discord-rpc/CMakeLists.txt (hunk 1 above)
Comment (p01arst0rm) [reply 3rdparty/discord-rpc/CMakeLists.txt:2]: sure :) should i append this as a commit to this pr?

Comment (hcorion) [reply 3rdparty/discord-rpc/CMakeLists.txt:2]: Yeah, sounds good to me

**Comment by hcorion** [comment]
Sorry, looks like I didn't explain as well as I should have 😅
The folder should still be called discord-rpc, just in the CMakeLists.txt is should be `add_library(DiscordRPC INTERFACE)`.
Which is similar to how upstream discord-rpc is setup.

**Comment by p01arst0rm** [comment]
fixed now :)

**Review by hcorion — APPROVED** [review]

**Pull request merged**